Geometry/HGCalMapping/plugins/alpaka/HGCalMappingCellESProducer.cc

```diff
@@ -41,9 +43,41 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
         desc.add<std::vector<std::string> >("filelist", std::vector<std::string>({}))
             ->setComment("list of files with the readout cells of each module");
         desc.add<edm::ESInputTag>("cellindexer", edm::ESInputTag(""))->setComment("Dense cell index tool");
+        desc.add<std::string>("offsetfile", std::string({}))
+            ->setComment("file containing the offsets between calibration and surrounding cells");
         descriptions.addWithDefaultLabel(desc);
       }
 
+      std::map<std::tuple<std::string, int, int>, int> makeOffsetMap(std::string input_offsetfile) {
+        std::map<std::tuple<std::string, int, int>, int> offsetMap;
+        auto offsetfile = edm::FileInPath(input_offsetfile).fullPath();
```

Comment:
It would be better to do the `FileInPath` in the constructor to catch errors from a missing file earlier. Actually you can use `FileInPath` as the configuration type, in which case errors would be caught even earlier.